### PR TITLE
Add EventType column for v1.2 in /event service.

### PR DESCRIPTION
**Pull request by junghao**
Regarding to https://github.com/GeoNet/tickets/issues/10574.
Added `EventType` column for compliance with FDSN event service v1.2.

The stations service has some unsupported query parameters like event service, but I'm not changing it to use a generalised error message because some services really using that query parameter - though the value is always the default value and requires no operation for existing query.

### CODE REVIEW

**Review by CallumNZ — APPROVED** [review]

File: cmd/fdsn-ws/fdsn_event.go [hunk 1]
```diff
@@ -176,6 +190,30 @@ func parseEventV1(v url.Values) (fdsnEventV1, error) {
...
+	if e.MagnitudeType != "" {
+		return e, errors.New("magnitudetype is not supported.")
+	}
+
+	if e.EventType != "" {
+		return e, errors.New("eventtype is not supported.")
+	}
+
+	if e.Limit != -1 {
+		return e, errors.New("ilimit is not supported.")
```
Comment (CallumNZ) [inline cmd/fdsn-ws/fdsn_event.go:202]: limit

**Review by CallumNZ — COMMENTED** [review]
one typo

**Review by CallumNZ — APPROVED** [review]

File: cmd/fdsn-ws/fdsn_event.go [hunk 2]
```diff
@@ -65,13 +68,21 @@ type Time struct {
...
+	"offset":               true,
+	"catalog":              true,
+	"contributor":          true,
+	"includeallorigins":    true,
+	"includeallmagnitudes": true,
+	"includearrivals":      true,
 }
 
+// copied from p.17 of FDSN-WS-Specifications-1.2.pdf
+const EVENT_TYPES = `not existing, not reported, earthquake, anthropogenic event, collapse, cavity collapse, mine collapse, building collapse, explosion, accidental explosion, chemical explosion, controlled explosion, experimental explosion, industrial explosion, mining explosion, quarry blast, road cut, blasting levee, nuclear explosion, induced or triggered event, rock burst, reservoir loading, fluid injection, fluid extraction, crash, plane crash, train crash, boat crash, other event, atmospheric event, sonic boom, sonic blast, acoustic noise, thunder, avalanche, snow avalanche, debris avalanche, hydroacoustic event, ice quake, slide, landslide, rockslide, meteorite, volcanic eruption`
```
Comment (CallumNZ) [inline cmd/fdsn-ws/fdsn_event.go:82]: Can this be made multiple lines?

File: cmd/fdsn-ws/fdsn_event.go [hunk 3]
```diff
@@ -238,9 +238,38 @@ func parseEventV1(v url.Values) (fdsnEventV1, error) {
...
+				qp += fmt.Sprintf("'%s',", t)
+			}
+		}
+
+		if qp != "" {
+			e.eventTypeStr = strings.TrimSuffix(qp, ",") // remove the unnecessary last comma
+		} else {
+			err = fmt.Errorf("invalid value for eventtype: %s", e.EventType)
+			return e, err
+		}
```
Comment (CallumNZ) [inline cmd/fdsn-ws/fdsn_event.go:266]: Suggestion:

```
if qp == "" {
    err = fmt.Errorf("invalid value for eventtype: %s", e.EventType)
    return e, err
}
e.eventTypeStr = strings.TrimSuffix(qp, ",") // remove the unnecessary last comma
```

File: cmd/fdsn-ws/routes.go [hunk 4]
```diff
@@ -57,3 +60,53 @@ func soh(r *http.Request, h http.Header, b *bytes.Buffer) error {
...
+		} else if strings.HasPrefix(e.url, "/fdsnws/station/") {
+			ver = stationVersion
+		} else if strings.HasPrefix(e.url, "fdsnws/dataselect/") {
+			ver = dataselectVersion
+		}
+
+		h.Set("Content-Type", "text/plain; charset=utf-8")
+
+		switch e.Code {
+		case http.StatusNoContent, http.StatusNotFound: // NOTE: thought NoContent is not an error but we handled here
```
Comment (CallumNZ) [inline cmd/fdsn-ws/routes.go:95]: *though ?

**Review by CallumNZ — CHANGES_REQUESTED** [review]

**Review by CallumNZ — APPROVED** [review]

**Review by CallumNZ — APPROVED** [review]
Looks good.

**Review by CallumNZ — APPROVED** [review]

**Comment by junghao** [comment]
@CallumNZ I've made an update regarding https://github.com/GeoNet/tickets/issues/12424#issuecomment-1427234143, could you please review? Thanks

**Review by CallumNZ — APPROVED** [review]
Looks good

**Pull request merged**